Libraries/Components/ActivityIndicator/ActivityIndicator.js

```diff
@@ -67,24 +67,37 @@ const ActivityIndicator = React.createClass({
   render() {
     const {onLayout, style, ...props} = this.props;
     let sizeStyle;
-    switch (props.size) {
-      case 'small':
-        sizeStyle = styles.sizeSmall;
-        break;
-      case 'large':
-        sizeStyle = styles.sizeLarge;
-        break;
+    let ReturnedActivityIndicator;
+    if (isNaN(props.size)) {
+      switch (props.size) {
+        case 'small':
+          sizeStyle = styles.sizeSmall;
+          break;
+        case 'large':
+          sizeStyle = styles.sizeLarge;
+          break;
+      }
+
+      ReturnedActivityIndicator = <RCTActivityIndicator
+                                  {...props}
+                                  style={sizeStyle}
+                                  styleAttr="Normal"
+                                  indeterminate
+                                 />;
+    } else {
+      ReturnedActivityIndicator = <RCTActivityIndicator
+                                  {...props}
+                                  style={ {height: props.size, width: props.size} }
+                                  styleAttr="Normal"
+                                  indeterminate
+                                 />;
     }
+
     return (
       <View
         onLayout={onLayout}
         style={[styles.container, style]}>
-        <RCTActivityIndicator
-          {...props}
-          style={sizeStyle}
-          styleAttr="Normal"
-          indeterminate
-        />
+        {ReturnedActivityIndicator}
```

Comment:
you don't need to conditionally set the RCTActivityIndicator. just setting the right sizeStyle variable should be enough